executor/writer.go

```diff
@@ -349,8 +347,22 @@ func WriteCSMInner(csm io.ColumnSeriesMap, isVariableLength bool) (err error) {
 			return fmt.Errorf(columnMismatchError, csDSV, dbDSV)
 		}
 		missing, coercion := GetMissingAndTypeCoercionColumns(dbDSV, csDSV)
-		if missing != nil || coercion != nil {
+		if missing != nil {
 			return fmt.Errorf(columnMismatchError, csDSV, dbDSV)
+		} else if coercion != nil {
```

Comment:
[nit] early return might be more readable
```suggestion
		}
		
		if coercion != nil {
```